lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1167,6 +1245,30 @@ class TextEditingChannel {
     window._replyToPlatformMessage(callback, codec.encodeSuccessEnvelope(true));
   }
 
+  /// Used for submitting the forms attached on the DOM.
+  ///
+  /// Browser will save the information entered to the form.
+  ///
+  /// Called when the form is finalized with save option `true`.
+  /// See: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/services/text_input.dart#L1277
+  void saveForms() {
+    formsOnTheDom.forEach((String identifier, html.FormElement form) {
+      final html.InputElement submitBtn =
+          form.getElementsByClassName('submitBtn').first as html.InputElement;
+      submitBtn.click();
+    });
+  }
+
+  /// Used for removing the forms on the DOM.
+  ///
+  /// Called when the form is finalized.
+  void cleanForms() {
+    while (html.document.getElementsByTagName('form').length > 0) {
+      html.document.getElementsByTagName('form').last.remove();
+    }
```

Comment:
This is too aggressive. It'll remove any form elements on the page, including ones that were created by someone else.

Is there a reason we are not using forms in the map `formsOnTheDom`?

Reply:
I changed it. I initially tried to avoid iterating over the map.

When I iterate over the maps, I need to use a cast like:
`final html.FormElement form = formsOnTheDom[formsOnTheDom.entries.last.key] as html.FormElement;`

Otherwise I get a non-nullability error such as `A value of type 'T?' can't be assigned to a variable of type 'T'`.